data_model/src/block.rs

```diff
@@ -136,24 +131,21 @@ mod committed {
         /// `VersionedCommitedBlock` should have the same hash as `VersionedCommitedBlock`.
         #[cfg(feature = "std")]
         #[inline]
-        pub fn hash(&self) -> HashOf<Self> {
-            self.as_v1().hash().transmute()
+        pub fn hash(&self) -> HashOf<BlockPayload> {
```

Comment:
pay attention to this change, the same is done for transactions. This makes it possible to interchangeably use hashes for different blocks/transactions without using `transmute`

Reply:
Clever